tools/DC_api_wrappers/DC_wrappers.py

```diff
@@ -0,0 +1,224 @@
+"""
+Wrappers for fetching schema from DC API.
+"""
+
+import ast
+import copy
+import json
+import logging
+import os
+import sys
+from absl import app
+from absl import flags
+from sys import path
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('dcid', None, 'dcid of the node to query')
+flags.DEFINE_string('dc_output_path', './prefetched_outputs/',
+                    'Path to store the output')
+flags.DEFINE_boolean('force_fetch', False,
+                     'forces api query and not return cached result')
+
+_MODULE_DIR = os.path.dirname(os.path.realpath(__file__))
+path.insert(1, os.path.join(_MODULE_DIR, '..'))
+
+from download_utils.requests_wrappers import request_post_json
+
+# logging.basicConfig() # you need to initialize logging, otherwise you will not see anything from requests
+# logging.getLogger().setLevel(logging.DEBUG)
+# requests_log = logging.getLogger("urllib3")
+# requests_log.setLevel(logging.DEBUG)
+# HTTPConnection.debuglevel = 1
+# requests_log.propagate = True
+
+
+def dc_check_existence(dcid_list: list,
+                       use_autopush: bool = True,
+                       max_items: int = 450) -> dict:
+    """Checks if a given list of dcids are present in DC.
+
+    Args:
+        dcid_list: List of dcids to be queried for existence.
+        use_autopush: Boolean value to use autopush API and not public API.
+        max_items: Limit of items to be queried in a single POST request.
+
+
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    data_ = {}
+    ret_dict = {}
+    if use_autopush:
+        url_prefix = 'autopush.'
+    else:
+        url_prefix = ''
+
+    chunk_size = max_items
+    dcid_list_chunked = [
+        dcid_list[i:i + chunk_size]
+        for i in range(0, len(dcid_list), chunk_size)
+    ]
+    for dcid_chunk in dcid_list_chunked:
+        data_["dcids"] = dcid_chunk
+        req = request_post_json(
+            f'https://{url_prefix}api.datacommons.org/node/property-labels',
+            data_)
+        resp_dicts = req['payload']
+        resp_dicts = ast.literal_eval(resp_dicts)
+        for cur_dcid in resp_dicts:
+            if not resp_dicts[cur_dcid]:
+                ret_dict[cur_dcid] = False
+            elif not resp_dicts[cur_dcid]['inLabels'] and not resp_dicts[cur_dcid]['outLabels']:
+                ret_dict[cur_dcid] = False
+            else:
+                ret_dict[cur_dcid] = True
+
+    return ret_dict
+
+# fetch pvs from dc, enums from dc
+
+def fetch_dcid_properties_enums(class_dcid: str,
+                                cache_path: str = _MODULE_DIR + '/prefetched_outputs',
+                                use_autopush: bool = True,
+                                force_fetch: bool = False):
+  """Fetches all the properties and it's possible values for a given dcid.
+
+    Args:
+      class_dcid: DCID of the object whose properties and enum values need to be fetched.
```

Comment:
could remove `class_` prefix since the object could be any node, not only `typeOf: Class`

Reply:
fixed.